src/gurt/telemetry.c

```diff
@@ -831,19 +854,46 @@ d_tm_init(int id, uint64_t mem_size, int flags)
 	return rc;
 }
 
+/* Check if all children are invalid */
+static bool
+is_node_empty(struct d_tm_node_t *node)
+{
+	struct d_tm_context   *ctx = tm_shmem.ctx;
+	struct d_tm_shmem_hdr *shmem;
+	struct d_tm_node_t    *child;
+
+	shmem = get_shmem_for_key(ctx, node->dtn_shmem_key);
+	child = conv_ptr(shmem, node->dtn_child);
```

Comment:
Just looking at the existing telemetry code, I noticed a few helpers already exist. `d_tm_get_child` and `d_tm_get_sibling` in particular. It may be a good idea to use these to clean up now, perhaps it will help to narrow down where the segfaults Dalton discovered are happening.